### PR TITLE
Make SiteWhitelist mandatory during assignment; remove site wildcards…

### LINKED ISSUE #7535: Make SiteWhitelist mandatory during assignment
and it cannot be an empty list

### PR DESCRIPTION
**Pull request by amaltaro**
…; updated cmsname Lexicon

Fixes #7535

Ideally, the validation should go under WMWorkloadTools, but I can't see how to distinguish creation from assignment and make it work with that special data type and the validate function. Seangchan, if you agree, I'd say we leave that up to the moment that we really separate creation from assignment arguments :-)

It passes all the basic tests.

### CODE REVIEW

**Comment by amaltaro** [comment]
@ticoann I still have to update the unit tests, but can you review it please?

File: src/python/WMCore/Lexicon.py [hunk 1]
```diff
@@ -127,15 +127,12 @@ def jobrange(candidate):
...
-        T2_UK_SGrid
-        T2_UK_SGrid_Bristol
+    Candidate must pass the CMS name pattern. Thus:
+     * good candidates: T2_UK_SGrid or T2_UK_SGrid_Bristol
+     * bad candidates: T2, T2_UK
     """
-    # remove any trailing _'s
     candidate = candidate.rstrip('_')
-    return check("^T[0-3%]((_[A-Z]{2}(_[A-Za-z0-9]+)*)?)$", candidate)
+    return check("^T[0-3]_[A-Z]{2}_[A-Za-z0-9]+", candidate)
```
Comment (amaltaro) [inline src/python/WMCore/Lexicon.py]: This regex is Ok but it's not Ok for the check method. I'm checking what should be the correct regex ... :(

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
tests are Ok in my VM (and agent)

**Review by ticoann — APPROVED** [review]

**Pull request merged**